azure-kusto-data/azure/kusto/data/client.py

```diff
@@ -853,19 +853,61 @@ def execute_streaming_ingest(
 
         self._execute(endpoint, database, None, stream, self._streaming_ingest_default_timeout, properties)
 
-    def _execute(
-        self, endpoint: str, database: str, query: Optional[str], payload: Optional[io.IOBase], timeout: timedelta, properties: ClientRequestProperties = None
+    def execute_streaming_query(

```

Comment:
Can't we just pass a boolean argument of 'streaming'?

Reply:
The return type of _execute is different than that of execute_streaming_query,
I think it's a bad form to change the return type of a function based on a parameter

Reply:
Same discussion as the result type